### PR TITLE
Dynamic protocol

### LINKED ISSUE #15: Protocol port of source config
Would you guys be opposed to being able to select a protocol as part of source config?

My issue right now is that I'm using a private docker repo that uses https (not http)

### PR DESCRIPTION
**Pull request by mehtaphysical**
resolves #15 

* set http/https depending on the availability of the v1/_ping endpoint
* add `-k` to curl_opts if `--insecure-registry` is set

### CODE REVIEW

**Comment by vito** [comment]
Sorry for the delay on this. Have you manually verified that this works? There was at least one point in time where the scripts used bash-specific features, but it looks like you've fixed it.

**Comment by mehtaphysical** [comment]
no worries.

It seems to be working. I've been using it without issue for a few weeks now. 

Yeah at first i wrote it as bash. I've updated it since then. Sorry about that.

File: assets/check [hunk 1]
```diff
@@ -35,23 +36,29 @@ fi
...
-curl ${curl_opts} "${uri_root}/v1/repositories/${repository}/images" > $all_images
+shortened_image=$(jq -r "try map(select(.name == $(echo $tag | jq -R .)))[].layer" < $tags)
 
-latest_image=$(jq -r "map(select(.id | startswith($(echo $shortened_image | jq -R .))))[0].id // \"\"" < $all_images)
+if [ -n shortend_image ]; then
+  all_images=$(mktemp /tmp/resource-in-tags.XXXXXX)
+  curl ${curl_opts} "${uri_root}/v1/repositories/${repository}/images" > $all_images
+  latest_image=$(jq -r "map(select(.id | startswith($(echo $shortened_image | jq -R .))))[0].id // \"\"" < $all_images)
+else
+  latest_image=$(jq -r "$(echo $tag) | jq -R .")
```
Comment (vito) [inline assets/check]: What are these changes for? Looks unrelated to the rest (also this line has broken syntax).

**Comment by vito** [comment]
So I ended up rewriting `check` to support v2 endpoints in 5d6c1ddfdf84e1799dc7866dfb9d7ec5fb2dfd54. Along the way I added an explicit `source.insecure_registries` property, and `check` will try `https` and fall back on `http`. Also `in` no longer does the `curl` so it shouldn't need anything special.

If you want to give it a try, you can point your instance at `docker:///concourse/docker-image-resource`. Note that a few things have changed (see the README). Notably it versions via digests now.
